src/api/schema/log_events/event.rs

```diff
@@ -0,0 +1,48 @@
+use crate::event;
+use async_graphql::Object;
+use chrono::{DateTime, Utc};
+use serde_json::json;
+
+#[derive(Debug)]
+pub struct LogEvent {
+    component_name: String,
+    event: event::LogEvent,
+}
+
+impl LogEvent {
+    pub fn new(component_name: &str, event: event::LogEvent) -> Self {
+        Self {
+            component_name: component_name.to_string(),
+            event,
+        }
+    }
+}
+
+#[Object]
+/// Log event with fields for querying log data
+impl LogEvent {
+    /// Name of the component associated with the log event
+    async fn component_name(&self) -> &str {
+        &self.component_name
+    }
+
+    /// Log message
+    async fn message(&self) -> Option<String> {
+        Some(self.event.get("message")?.to_string_lossy())
+    }
+
+    /// Log timestamp
+    async fn timestamp(&self) -> Option<&DateTime<Utc>> {
+        self.event.get("timestamp")?.as_timestamp()
+    }
+
+    /// Log event as a JSON string
+    async fn json(&self) -> String {
+        json!(self.event).to_string()
```

Comment:
I think we should be using `serde_json::to_string()` here. `json!` will panic if it cannot serialize. That macro seems to be more for inlining JSON literals than serialization of types.

Reply:
Is there a scenario in which an `event::LogEvent` wouldn't serialize to JSON?

I assumed this was infallible, returning a `String!` vs. `String` to avoid the client having to unwrap, but this could well be a false assumption.

Reply:
Yeah, there shouldn't be a situation where that happens. I think `serde_json::to_string(self.event).expect("will always serialize")` would make this more readily apparent. You'll find this is how we do it in sinks for example:

https://github.com/timberio/vector/blob/0e39cacd7de327d20c48bc156103b842064f2427/src/sinks/http.rs#L199-L205

I think the same should be true of YAML.